packages/orama/src/components/groups.ts

```diff
@@ -168,7 +168,11 @@ function calculateCombination(arrs: ScalarSearchableValue[][], index = 0): Scala
   const combinations = []
   for (const value of head) {
     for (const combination of c) {
-      combinations.push([value, ...combination])
+      const result = [value];
+
+      safeArrayPush(result, combination)
+
+      combinations.push(result)
```

Comment:
I fear I missed something; do we need both `safeArrayPush(result, combination)` and `combinations.push(result)` here? Otherwise, LGTM

Reply:
Yes, because `result` is just `[value, ...combination]`.

So we need to push it.